### PR TITLE
Simplify log file manager hooks to one hook

### PR DESCRIPTION
**Pull request by ffried**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: #1165 

### Pull Request Description

Currently, there are two hooks in DDLogFileManager for managing archived log files (`didRollAndArchiveLogFile:` and `didArchiveLogFile:`). The only difference is the place where they are called.
This simplifies that by turning the two methods into one with an additional bool parameter that provides that distinction.

### CODE REVIEW

**Review by sushichop — APPROVED** [review]

File: Documentation/LogFileManagement.md [hunk 1]
```diff
@@ -47,20 +47,19 @@ Let's take a look at the DDLogFileManager protocol:
...
-- (NSString *)createNewLogFile;
+- (NSString *)createNewLogFileWithError(NSError**)error;
 
 @optional
 
 // Notifications from DDFileLogger
 
-- (void)didArchiveLogFile:(NSString *)logFilePath;
-- (void)didRollAndArchiveLogFile:(NSString *)logFilePath;
+- (void)didRollAndArchiveLogFile:(NSString *)logFilePath wasRolled:(BOOL)wasRolled;
```
Comment (imran20487) [inline Documentation/LogFileManagement.md:56]: Correction: (void)**didArchiveLogFile**:(NSString *)logFilePath wasRolled:(BOOL)wasRolled;

File: Sources/CocoaLumberjack/DDFileLogger.m [hunk 2]
```diff
@@ -1069,11 +1083,23 @@ - (BOOL)lt_shouldUseLogFile:(nonnull DDLogFileInfo *)logFileInfo isResuming:(BOO
...
+        const BOOL logFileManagerRespondsToNewArchiveSelector = [_logFileManager respondsToSelector:@selector(didArchiveLogFile:wasRolled:)];
+        if (logFileManagerRespondsToNewArchiveSelector || [_logFileManager respondsToSelector:@selector(didArchiveLogFile:)]) {
             NSString *archivedFilePath = [logFileInfo.filePath copy];
-            dispatch_async(_completionQueue, ^{
-                [self->_logFileManager didArchiveLogFile:archivedFilePath];
-            });
+            dispatch_block_t block;
+            if (logFileManagerRespondsToNewArchiveSelector) {
+                block = ^{
+                    [self->_logFileManager didArchiveLogFile:archivedFilePath wasRolled:YES];
```
Comment (imran20487) [inline Sources/CocoaLumberjack/DDFileLogger.m:1092]: Shouldn't this be `wasRolled:NO` ?

File: Sources/CocoaLumberjack/include/CocoaLumberjack/DDFileLogger.h [hunk 3]
```diff
@@ -161,16 +161,22 @@ extern unsigned long long const kDDDefaultLogFilesDiskQuota;
 
 // Notifications from DDFileLogger
 
+/// Called when a log file was archived. Executed on global queue with default priority.
+/// @param logFilePath TThe path to the log file that was archived.
```
Comment (imran20487) [inline Sources/CocoaLumberjack/include/CocoaLumberjack/DDFileLogger.h:165]: typo: a double T in The

File: Documentation/LogFileManagement.md (hunk 1 above)
Comment (ffried) [reply Documentation/LogFileManagement.md]: Ah, copied the wrong method. Thanks!

File: Sources/CocoaLumberjack/include/CocoaLumberjack/DDFileLogger.h (hunk 3 above)
Comment (ffried) [reply Sources/CocoaLumberjack/include/CocoaLumberjack/DDFileLogger.h]: Fixed...

File: Sources/CocoaLumberjack/DDFileLogger.m (hunk 2 above)
Comment (ffried) [reply Sources/CocoaLumberjack/DDFileLogger.m]: Totally, good catch, thanks!

Comment (imran20487) [reply Sources/CocoaLumberjack/DDFileLogger.m]: In this case, the file is only archived, so `wasRolled` should be `NO`

**Review by imran20487 — CHANGES_REQUESTED** [review]

**Pull request merged**